src/main/java/com/xatkit/core/recognition/processor/PerspectiveapiInterface.java

```diff
@@ -0,0 +1,281 @@
+         */
+        LIKELY_TO_REJECT,
+        /**
+         * Obscene or vulgar language such as cursing.
+         */
+        OBSCENE,
+        /**
+         * Irrelevant and unsolicited commercial content.
+         */
+        SPAM,
+        /**
+         * Trivial or short comments.
+         */
+        UNSUBSTANTIAL;
+    }
+
+    private String apiKey;
+    private String commentText;
+    private String commentType = "PLAIN_TEXT";
+    private AttributeType[] requestedAttributes;
+    private String language;
+    private Boolean doNotStore;
+    private String clientToken;
+    private String sessionId;
+    private JSONObject request;
+    private Map<String, AttributeType[]> languageAttributes;
+    private String attributesPrefix;
+
+    /**
+     * Constructor
+     *
+     * @param apiKey           the api key
+     * @param language         the language
+     * @param doNotStore       if true then data will not be stored to PerspectiveAPI. Default is false
+     * @param clientToken      the client token
+     * @param sessionId        the session id
+     * @param attributesPrefix the attributes prefix for the keys of the map they will be stored in
+     */
+    PerspectiveapiInterface(@NonNull String apiKey, String language, Boolean doNotStore, String clientToken,
+                            String sessionId, String attributesPrefix) {
+
+        this.apiKey = apiKey;
+        this.language = language != null ? language : "en";
+        this.doNotStore = doNotStore != null ? doNotStore : false;
+        this.clientToken = clientToken;
+        this.sessionId = sessionId;
+        this.attributesPrefix = attributesPrefix;
+        languageAttributes = new HashMap<>();
+        languageAttributes.put("en", new AttributeType[] {
+                AttributeType.TOXICITY,
+                AttributeType.SEVERE_TOXICITY,
+                AttributeType.TOXICITY_FAST,
+                AttributeType.IDENTITY_ATTACK,
+                AttributeType.INSULT,
+                AttributeType.PROFANITY,
+                AttributeType.THREAT,
+                AttributeType.SEXUALLY_EXPLICIT,
+                AttributeType.FLIRTATION,
+                AttributeType.ATTACK_ON_AUTHOR,
+                AttributeType.ATTACK_ON_COMMENTER,
+                AttributeType.INCOHERENT,
+                AttributeType.INFLAMMATORY,
+                AttributeType.LIKELY_TO_REJECT,
+                AttributeType.OBSCENE,
+                AttributeType.SPAM,
+                AttributeType.UNSUBSTANTIAL
+        });
+        languageAttributes.put("es", new AttributeType[] {
+                AttributeType.TOXICITY,
+                AttributeType.SEVERE_TOXICITY,
+                AttributeType.IDENTITY_ATTACK_EXPERIMENTAL,
+                AttributeType.INSULT_EXPERIMENTAL,
+                AttributeType.PROFANITY_EXPERIMENTAL,
+                AttributeType.THREAT_EXPERIMENTAL
+        });
+        this.setRequestedAttributes();
```

Comment:
This is very confusing: the method returns a value, but it is ignored. Yet one of the class attribute has been updated by a side effect.
Since the requested attributes cannot change (the language is initialized once when the client is constructed) we can remove the `setRequestedAttributes` method and directly compute its result in the constructor.